modulemd/v2/modulemd-profile.c

```diff
@@ -46,6 +46,29 @@ enum
 static GParamSpec *properties[N_PROPS];
 
 
+gboolean
+modulemd_profile_equals (ModulemdProfile *self_1, ModulemdProfile *self_2)
+{
+  g_return_val_if_fail (MODULEMD_IS_PROFILE (self_1), FALSE);
+  g_return_val_if_fail (MODULEMD_IS_PROFILE (self_2), FALSE);
+
+  if (g_strcmp0 (modulemd_profile_get_name (self_1),
+                 modulemd_profile_get_name (self_2)) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (modulemd_profile_get_description (self_1, NULL),
+                 modulemd_profile_get_description (self_2, NULL)))
+    return FALSE;
+
+  //Check rpms: size, keys, values
+  if (!modulemd_hash_table_equals (
```

Comment:
You don't need to call `modulemd_hash_table_equals()` here, because `self->rpms` are already just sets. You can call `modulemd_hash_table_sets_are_equal (self_1->rpms, self_2->rpms)` directly here.